src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/src/components/api-authorization/AuthorizeService.js

```diff
@@ -310,6 +310,10 @@ export class AuthorizeService {
         });
 
         this.userManager = new UserManager(settings);
+        this.userManager.events.addUserSignedOut(async () => {
+            await this.userManager.removeUser();
+            this.updateState(undefined);
```

Comment:
I don't really know oidc-client well enough to know what this really does (e.g., `this.userManager.removeUser`), so I can only assume it's correct and has the desired behaviour. Same with the Angular version.

Reply:
Yeah, that just clears up the state from the local user cache. And then we trigger an update to the user. That makes it so that when you log out from Identity, the SPA also gets notified and updates its state.